### ISSUE TITLE
Improve snipe inspectors

### ISSUE THREAD

**Issue by mars0i** [mars0i/free-agent#23]
Right now I can inspect a snipe, but the data doesn't update, and I can't place a circle around it using CircledPortrayal2D, etc.  The former might be due to the fact that I keep replacing snipes, FP-style, but the latter seems different.  I can't even put a snipe portrayal in a *selected* state, it seems.  Maybe this is the problem with updating, too.  Note that in my Students Clojure sim, this all works, and I didn't have to do any explicit calls to `setVolatile` on the agents, or anything like that.  Note though that I used `setPortrayalForAll` there, and I don't in free-agent.  Is that relevant?

**Comment by mars0i** [comment]
The problem with `CircledPortrayal2D` is an interaction with `HexaObjectGridPortrayal2D`.  It doesn't happen with `ObjectGridPortrayal2D`.  See masonborder repo.  I asked about it on the mailing list.

**Comment by mars0i** [comment]
Maybe the inspector doesn't update because the inspector system actually locks on to the agent itself, rather than querying whatever's in that place in the field portrayal or something like that.  But I keep changing the actual object.  

So maybe what I need to do is to write an inspector subclass that will query snipes by id or something like that.  

**But first** perform this experiment: Temporarily take out all of the ObjectGrid2D cloning from popenv.clj and just pass around a single Grid2D for snipe-field.  See if that solves the problem.

**Comment by mars0i** [comment]
Problem was due to functional updating, with new snipe objects on each tick representing the same logical snipe.  My solution was have the snipes implement `Propertied`, providing a `properties` method that returned a `SimpleProperty` subclass using `proxy`.  This didn't work initially, because snipes are defrecords and defrecords implement `java.util.Map`, which means that they are collections.  Therefore MASON wanted to handle them with `CollectionProperties` rather than `SimpleProperties, and only the latter supported `Propertied` or `Proxyiable` (another strategy I tried at one point--but `Propertied` was conceptually simpler).  Sean Luke very nicely modified Properties.java to allow `Propertied` to work with collections, and now I have working snipe inspectors. (!)
